docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Expression.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic.Transforms
+{
+    public static class Expression
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for
+            // exception tracking and logging, as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Create a small dataset as an IEnumerable.
+            var samples = new List<InputData>()
+            {
+                new InputData(0.5f, new[] { 1f, 0.2f }, 3, "hi", true, new[] { "zero", "one" }),
+                new InputData(-2.7f, new[] { 3.5f, -0.1f }, 2, "bye", false, new[] { "a", "b" }),
+                new InputData(1.3f, new[] { 1.9f, 3.3f }, 39, "hi", false, new[] { "0", "1" }),
+                new InputData(3, new[] { 3f, 3f }, 4, "hello", true, new[] { "c", "d" }),
+                new InputData(0, new[] { 1f, 1f }, 1, "hi", true, new[] { "zero", "one" }),
+                new InputData(30.4f, new[] { 10f, 4f }, 9, "bye", true, new[] { "e", "f" }),
+                new InputData(5.6f, new[] { 1.1f, 2.2f }, 0, "hey", false, new[] { "g", "h" }),
+            };
+
+            // Convert training data to IDataView.
+            var dataview = mlContext.Data.LoadFromEnumerable(samples);
+
+            // A pipeline that applies various expressions to the input columns.
+            var pipeline = mlContext.Transforms.Expression("Expr1", "(x,y)=>log(y)+x", "Float", "FloatVector")
+                .Append(mlContext.Transforms.Expression("Expr2", "(b,s,i)=>b ? len(s) : i", "Boolean", "StringVector", "Int"))
```

Comment:
When first reading this I thought (incorrectly) we had to note the data types for inputs/outputs. Can we use non-datatypes (eg. `HasSiblings`) or longer names (eg. `BooleanExampleColumn`)?
```suggestion
                .Append(mlContext.Transforms.Expression("Expr2", "(b,s,i)=>b ? len(s) : i", "HasSiblings", "PreferredGreetings", "Age"))
```

Reply:
Thanks for fixing
#resolved